packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -15,72 +15,113 @@
  * limitations under the License.
  */
 
+import { deepEqual } from '@firebase/util';
+
 import { Value } from '../protos/firestore_proto_api';
 import { invokeRunAggregationQueryRpc } from '../remote/datastore';
 import { hardAssert } from '../util/assert';
 import { cast } from '../util/input_validation';
 
 import { getDatastore } from './components';
 import { Firestore } from './database';
-import { Query, queryEqual } from './reference';
+import { DocumentFieldValue, Query, queryEqual } from './reference';
 import { LiteUserDataWriter } from './reference_impl';
 
 /**
- * An `AggregateQuery` computes some aggregation statistics from the result set of
- * a base `Query`.
+ * An `AggregateField` computes some aggregation statistics from the result set of
+ * an aggregation query.
  */
-export class AggregateQuery {
-  readonly type = 'AggregateQuery';
-  /**
-   * The query on which you called `countQuery` in order to get this `AggregateQuery`.
-   */
-  readonly query: Query<unknown>;
+export class AggregateField<T> {
+  type = 'AggregateField';
+  _datum?: T;
 
-  /** @hideconstructor */
-  constructor(query: Query<unknown>) {
-    this.query = query;
-  }
+  constructor(readonly subType: string) {}
+}
+
+/**
+ * The union of all `AggregateField` types that are returned from the factory
+ * functions.
+ */
+export type AggregateFieldType =
```

Comment:
I found a more elegant way to express AggregateFieldType using TypeScript's ReturnType type modifier. See https://github.com/dconeybe/CountApi/commit/7786142ca734b7def93ba6bd12d1c102db109866 and make that change here as well. HOWEVER, just include the one for count, and leave out the one for sum and min.